applications/API/permissions.py

```
@@ -0,0 +1,21 @@
+
+"""
+
+from rest_framework import permissions
+
+
+class IsOwnerOrReadOnly(permissions.BasePermission):
+    """
+    Custom permission to only allow owners of an insult to edit/delete it.
+    """
+
+    def has_object_permission(self, request, view, obj):
```

Ruff (0.12.2)

Missing return type annotation for public function `has_object_permission`

(ANN201)